### PR TITLE
chore: update gosec

### PR DESCRIPTION
**Pull request by gartnera**
# Description

All my changes were merged upstream. Continue using our fork to avoid docker hub rate limiting. We really should just use the gosec plugin on golangci-lint.

Main impact is a continued reduction in G115 false-negatives and false-positives. We will need to review the new errors before merge.

## Summary by CodeRabbit

- **New Features**
	- Enhanced P2P diagnostics with periodic peer discovery and communication.
	- Improved transaction receipt checking for Bitcoin transactions.
	- Updated ticker interval handling for inbound and outbound observers.

- **Bug Fixes**
	- Enhanced error handling and logging across various methods for better clarity and robustness.

- **Documentation**
	- Added security comments to clarify the handling of positive values and improve code readability.

- **Chores**
	- Updated dependencies for security scanning tools and workflows.

### CODE REVIEW

File: scripts/gosec.sh [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #!/usr/bin/env bash
 
-docker run -it --rm -w /node -v "$(pwd):/node" ghcr.io/zeta-chain/gosec:2.21.0-zeta -exclude-generated -exclude-dir testutil ./...
\ No newline at end of file
+docker run -it --rm -w /node -v "$(pwd):/node" ghcr.io/zeta-chain/gosec:2.21.4-zeta2 -exclude-generated -exclude-dir testutil ./...
```
Comment (swift1337) [inline scripts/gosec.sh:3]: nit: new line

File: zetaclient/orchestrator/orchestrator.go [hunk 2]
```diff
@@ -618,7 +618,7 @@ func (oc *Orchestrator) ScheduleCctxSolana(
 		oc.logger.Error().Msgf("ScheduleCctxSolana: chain observer is not a solana observer")
 		return
 	}
-	// #nosec G701 positive
+	// #nosec G115 G701 positive
```
Comment (swift1337) [inline zetaclient/orchestrator/orchestrator.go:621]: typo?

**Review by swift1337 — APPROVED** [review]

**Review by fbac — APPROVED** [review]

File: zetaclient/orchestrator/orchestrator.go (hunk 2 above)
Comment (gartnera) [reply zetaclient/orchestrator/orchestrator.go:621]: no, but it is incorrect. G701 was removed. Updated.

File: pkg/ticker/ticker.go [hunk 3]
```diff
@@ -183,7 +183,8 @@ func (t *Ticker) Stop() {
 	t.logger.Info().Msgf("Ticker stopped")
 }
 
-// SecondsFromUint64 converts uint64 to time.Duration in seconds.
-func SecondsFromUint64(d uint64) time.Duration {
-	return time.Duration(d) * time.Second
+// DurationFromUint64Seconds converts uint64 of secodns to time.Duration.
```
Comment (CryptoFewka) [inline pkg/ticker/ticker.go:186]: nitpick: typo of "secodns" 
`:%s/secodns/seconds/g`

**Review by CryptoFewka — APPROVED** [review]

**Pull request merged**